python/whylogs/core/validators/validator.py

```diff
@@ -1,19 +1,40 @@
+import copy
 from abc import ABC, abstractmethod
-from typing import Any, Callable, Dict, List
+from typing import Any, Callable, Dict, List, Optional, Union
 
 from whylogs.core.metrics.condition_count_metric import Condition
-from whylogs.core.preprocessing import PreprocessedColumn
 
 
 class Validator(ABC):
     name: str
     conditions: Dict[str, Condition]
-    actions: List[Callable[[str, str, Any], None]]
+    actions: List[Union[Callable[[str, str, Any], None], Callable[[str, str, Any, Optional[Any]], None]]]
 
     @abstractmethod
-    def columnar_validate(self, data: PreprocessedColumn):
+    def columnar_validate(self, data: Any, identity_values: Any = None) -> None:
```

Comment:
```suggestion
    def columnar_validate(self, data: Any, identity_values: Optional[Any] = None) -> None:
```
Why is `data` no longer a `PreprocessedColumn`?

Reply:
A PreprocessedColumn won't preserve the original order, and that's a problem when we want to correlate the values with the identity PreprocessedColumn, so it needs to be done on the raw data. This is not so much of a problem for the track_datum case, but it is for dataframes.

Reply:
I still think `identity_values` needs to be `Optional`

Reply:
right, missed that